sdk/include/opentelemetry/sdk/logs/logger.h

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "opentelemetry/logs/log_record.h"
+#include "opentelemetry/logs/logger.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/sdk/common/atomic_shared_ptr.h"
+#include "opentelemetry/sdk/logs/processor.h"
+
+#include <unordered_map>
+#include <vector>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+class Logger final : public opentelemetry::logs::Logger, public std::enable_shared_from_this<Logger>
+{
+public:
+  /**
+   * Initialize a new logger with a specified name.
+   * @param name The name of this logger.
```

Comment:
Do we have this param?

Reply:
I'm also curious what happened to the name parameter. In log4cxx - you'd have this:

```cpp
    static LoggerPtr getLogger(const std::string& name)
```

If we have this, we'd provide 1-1 mapping, easy-to-implement projection from log4cxx to OT logger. You can tell that the logger name is an identifier: either class name, or module name, or module / component tag.. It could be up to exporter to decide what to do with this name, but having it, in my opinion, is a good thing.

Reply:
Originally the logger constructor took in the name of the logger instance and it would be stored internally inside the logger. The `@param name` comment was left in by accident so we can remove that. 

The reason we removed the name parameter is that we chose to use a map inside the LoggerProvider to store the list of Loggers, with the key being the name. We used the logger name simply as a way to register the logger to the logger provider. If the logger also contained its name then it would waste memory, since it would be stored twice. Yea I agree with you Max that it could be useful for the logger to inject its name into the logs for contextual information. I'm thinking for now we leave it as is, and in the future re-add it if required?

Reply:
I'm OK to keep as-is, just clean the comment. We can discuss later. This is not a show stopper.
You can mark this conversation as resolved.